lambdas/src/apis/collections/controllers/collections.ts

```diff
@@ -11,92 +14,135 @@ import {
-      }, [])
-
-      const standardErc721 = {
-        id: urn,
-        name,
-        description,
-        language: 'en-US',
-        image,
-        thumbnail,
-        attributes: [
-          {
-            trait_type: 'Rarity',
-            value: wearableMetadata.rarity
-          },
-          {
-            trait_type: 'Category',
-            value: wearableMetadata.data.category
-          },
-          ...tagTraits,
-          ...bodyShapeTraits
-        ]
-      }
-      res.send(standardErc721)
-    } else {
-      res.status(404).send()
+        return bodyShapes
+      },
+      []
+    )
+
+    const tagTraits = wearableMetadata.data.tags.reduce((tags: ERC721StandardTrait[], tag) => {
+      tags.push({
+        trait_type: 'Tag',
+        value: tag
+      })
+
+      return tags
+    }, [])
+
+    const standardErc721 = {
+      id: urn,
+      name,
+      description,
+      language: 'en-US',
+      image,
+      thumbnail,
+      attributes: [
+        {
+          trait_type: 'Rarity',
+          value: wearableMetadata.rarity
+        },
+        {
+          trait_type: 'Category',
+          value: wearableMetadata.data.category
+        },
+        ...tagTraits,
+        ...bodyShapeTraits
+      ]
     }
+    res.send(standardErc721)
   } catch (e) {
-    res.status(500).send(e.message)
+    res.status(e.statusCode ?? 500).send(e.message)
   }
 }
 
-export async function contentsImage(client: SmartContentClient, req: Request, res: Response): Promise<void> {
+export async function contentsThumbnail(
+  client: SmartContentClient,
+  req: Request,
+  res: Response,
+  storage: string
+): Promise<void> {
   // Method: GET
-  // Path: /contents/:urn/image
-  const { urn } = req.params
-  await internalContents(client, res, urn, (wearableMetadata) => wearableMetadata.image)
+  // Path: /contents/:urn/thumbnail/:size
+  await handleImageRequest(client, req, res, storage, true)
 }
 
-export async function contentsThumbnail(client: SmartContentClient, req: Request, res: Response): Promise<void> {
+export async function contentsImage(
+  client: SmartContentClient,
+  req: Request,
+  res: Response,
+  storage: string
+): Promise<void> {
   // Method: GET
-  // Path: /contents/:urn/thumbnail
+  // Path: /contents/:urn/image/:size
+  await handleImageRequest(client, req, res, storage, false)
+}
+
+async function handleImageRequest(
+  client: SmartContentClient,
+  req: Request,
+  res: Response,
+  rootStorageLocation: string,
+  isThumbnail: boolean
+): Promise<void> {
   const { urn } = req.params
+  const size = req.params.size ?? DEFAULT_IMAGE_SIZE
+
+  try {
+    validateSize(size)
 
-  await internalContents(client, res, urn, (wearableMetadata) => wearableMetadata.thumbnail)
+    const entity = await fetchEntity(client, urn)
+    const imageRequest = {
+      urn,
+      hash: getFileHash(entity, entity.metadata.thumbnail),
+      size,
+      rarity: isThumbnail ? undefined : entity.metadata?.rarity
```

Comment:
nit: when it's a thumbnail it's not that the rarity is undefined but you are not using it, maybe we should add another field for that?

Reply:
yes, i should probably rename this property to something like `background`